pilz_trajectory_generation/include/pilz_trajectory_generation/tip_frame_getter.h

```diff
@@ -0,0 +1,39 @@
+#ifndef TIP_FRAME_GETTER_H
+#define TIP_FRAME_GETTER_H
+
+#include <string>
+#include <cassert>
+
+#include <moveit/robot_model/joint_model_group.h>
+
+#include "pilz_trajectory_generation/trajectory_generation_exceptions.h"
+
+namespace pilz_trajectory_generation
+{
+
+CREATE_MOVEIT_ERROR_CODE_EXCEPTION(EndEffectorException, moveit_msgs::MoveItErrorCodes::FAILURE);
+CREATE_MOVEIT_ERROR_CODE_EXCEPTION(NoSolverException, moveit_msgs::MoveItErrorCodes::FAILURE);
+
+/**
+ * @return The name of the tip frame (link) of the specified group. An
+ * exception is thrown if the group is an end-effector or if no solver for
+ * the group could be found.
+ */
+static const std::string& getTipFrame(const moveit::core::JointModelGroup* group)
+{
+  if (group->isEndEffector())
```

Comment:
Why is the end-effector check necessary? The pg70 gripper does not have an ik solver specified, so it would throw few lines below (currently line 32).
The point here is, that blending needs an solver (see line 30 below). Why shouldn't it be possible to get the tip link of an end-effector (the purpose of this function according to its name)?

Reply:
I agree, see #144

Reply:
Improvement, see #153